frontends/common/options.cpp

```diff
@@ -32,6 +32,21 @@ const char* p4_14includePath = CONFIG_PKGDATADIR "/p4_14include";
 
 const char* CompilerOptions::defaultMessage = "Compile a P4 program";
 
+template <size_t N>
+static void convertToAbsPath(const char* const relPath, char (&output)[N]) {
+    output[0] = '\0';  // Default to the empty string, indicating failure.
+
+    char cwd[PATH_MAX];
+    getcwd(cwd, sizeof(cwd));
+    const size_t cwdLen = strlen(cwd);
+    if (cwdLen == 0) return;
+    const char* separator = cwd[cwdLen - 1] == '/' ? "" : "/";
+
+    // Construct an absolute path. We're assuming that @relPath is relative to
+    // the current working directory.
+    snprintf(output, N, "%s%s%s", cwd, separator, relPath);
+}
+
```

Comment:
because realpath will follow the symbolic name of p4c-bm2-ss in build/p4c to `build/backends/bmv2/p4c-bm2-ss`, and `p4includePath` will be generated as `build/backends/bmv2/p4c-bm2-ss/p4include`, instead of `/build/p4c/p4include`. I used p4includePath as prefix to match on the path to system p4 files, which wouldn't work.